Support for Laravel 5.5

cappuc opened the pull request:
Hi,
this PR add support for Laravel 5.5.
I started from branch 1.4 because it was the only with support for Laravel 5.4

I made the changes listed here #239

Mulkave commented:
this is amazing! Thank you @cappuc 

one question before merging, does it break compatibility with any previous Laravel version? if so, we'll have to explicitly mention that in the docs for the supported versions.

cappuc commented:
Yes, it break compatibility with previous Laravel version because I updated Illuminate package versions to 5.5 and there are some methods signature changes in the Iluminate Model.

I also updated the php version required to 7.0 because Laravel 5.5 require it and I updated Travis CI tests:
- Removed php 5.6
- Added php 7.1
- Test on multiple NEO4J versions
